### PR TITLE
[logging] Add centralized logging - Bump-up cache loads to warnings

### PR DESCRIPTION
**Pull request by thomwolf**
Add a `nlp.logging` module to set the global logging level easily. The verbosity level also controls the tqdm bars (disabled when set higher than INFO).

You can use:
```
nlp.logging.set_verbosity(verbosity: int)
nlp.logging.set_verbosity_info()
nlp.logging.set_verbosity_warning()
nlp.logging.set_verbosity_debug()
nlp.logging.set_verbosity_error()
nlp.logging.get_verbosity() -> int
```
And use the levels:
```
nlp.logging.CRITICAL
nlp.logging.DEBUG
nlp.logging.ERROR
nlp.logging.FATAL
nlp.logging.INFO
nlp.logging.NOTSET
nlp.logging.WARN
nlp.logging.WARNING
```

### CODE REVIEW

File: benchmarks/results/benchmark_array_xd.json [hunk 1]
```diff
@@ -1 +1 @@
...mpy after write_array2d": 0.018853643006877974, "read_col_unformated after write_array2d": 0.05644163000397384, "read_col_formatted_as_numpy after write_array2d": 0.011610292000113986, "write_nested_sequence": 1.6535991109994939, "read_unformated after write_nested_sequence": 0.3739209540071897, "read_formatted_as_numpy after write_nested_sequence": 0.40762836500653066, "read_batch_unformated after write_nested_sequence": 0.3337586460111197, "read_batch_formatted_as_numpy after write_nested_sequence": 0.054717567007173784, "read_col_unformated after write_nested_sequence": 0.3173944180016406, "read_col_formatted_as_numpy after write_nested_sequence": 0.004956340009812266, "write_flattened_sequence": 1.4975415869994322, "read_unformated after write_flattened_sequence": 0.26713552299770527, "read_formatted_as_numpy after write_flattened_sequence": 0.07673935199272819, "read_batch_unformated after write_flattened_sequence": 0.25450974798877724, "read_batch_formatted_as_numpy after write_flattened_sequence": 0.009374254994327202, "read_col_unformated after write_flattened_sequence": 0.25912448299641255, "read_col_formatted_as_numpy after write_flattened_sequence": 0.004277604995877482}
```
Comment (lhoestq) [inline benchmarks/results/benchmark_array_xd.json:1]: are these files supposed to be part of the PR ?

File: src/nlp/arrow_dataset.py [hunk 2]
```diff
@@ -1589,8 +1572,7 @@ def shuffle(
                 }
                 indices_cache_file_name = self._get_cache_file_path(self.shuffle, cache_kwargs)
             if os.path.exists(indices_cache_file_name) and load_from_cache_file:
-                if verbose:
-                    logger.info("Loading cached shuffled indices for dataset at %s", indices_cache_file_name)
+                logger.warn("Loading cached shuffled indices for dataset at %s", indices_cache_file_name)
```
Comment (lhoestq) [inline src/nlp/arrow_dataset.py:1575]: use `logger.warning` instead ? iirc `warn` is deprecated

File: src/nlp/arrow_dataset.py [hunk 3]
```diff
@@ -1762,12 +1741,11 @@ def train_test_split(
                 and os.path.exists(test_indices_cache_file_name)
                 and load_from_cache_file
             ):
-                if verbose:
-                    logger.info(
-                        "Loading cached split indices for dataset at %s and %s",
-                        train_indices_cache_file_name,
-                        test_indices_cache_file_name,
-                    )
+                logger.warn(
```
Comment (lhoestq) [inline src/nlp/arrow_dataset.py:1744]: same here

File: src/nlp/arrow_dataset.py [hunk 4]
```diff
@@ -1509,8 +1496,7 @@ def sort(
                 }
                 indices_cache_file_name = self._get_cache_file_path(self.sort, cache_kwargs)
             if os.path.exists(indices_cache_file_name) and load_from_cache_file:
-                if verbose:
-                    logger.info("Loading cached sorted indices for dataset at %s", indices_cache_file_name)
+                logger.warn("Loading cached sorted indices for dataset at %s", indices_cache_file_name)
```
Comment (lhoestq) [inline src/nlp/arrow_dataset.py:1499]: same here

File: src/nlp/arrow_dataset.py [hunk 5]
```diff
@@ -1112,8 +1112,7 @@ def apply_function_on_filtered_inputs(inputs, indices, check_same_num_examples=F
                 }
                 cache_file_name = self._get_cache_file_path(function, cache_kwargs)
             if os.path.exists(cache_file_name) and load_from_cache_file:
-                if verbose:
-                    logger.info("Loading cached processed dataset at %s", cache_file_name)
+                logger.warn("Loading cached processed dataset at %s", cache_file_name)
```
Comment (lhoestq) [inline src/nlp/arrow_dataset.py:1115]: same here

File: src/nlp/utils/logging.py [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    Note that log propagation is disabled by default.
+    """
+
+    _configure_library_root_logger()
+    _get_library_root_logger().propagate = False
+
+
+def enable_propagation() -> None:
+    """Enable propagation of the library log outputs.
+    Please disable the HuggingFace Transformers's default handler to prevent double logging if the root logger has
```
Comment (lhoestq) [inline src/nlp/utils/logging.py:186]: what would be the issue with transformers exactly ?

**Review by lhoestq — APPROVED** [review]
Nice addition !

File: benchmarks/results/benchmark_array_xd.json (hunk 1 above)
Comment (thomwolf) [reply benchmarks/results/benchmark_array_xd.json:1]: we don't care that much I guess but let me remove them indeed

File: src/nlp/utils/logging.py (hunk 6 above)
Comment (thomwolf) [reply src/nlp/utils/logging.py:186]: copy-past error

**Pull request merged**